### PR TITLE
Check version

### PR DESCRIPTION
**Pull request by Maithem**

### CODE REVIEW

File: runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java [hunk 1]
```diff
@@ -538,6 +541,28 @@ public synchronized CorfuRuntime connect() {
...
+            CompletableFuture<VersionInfo>[] futures = layout.get().getLayoutServers()
+                    .stream().map(this::getRouter)
+                    .map(r -> r.getClient(BaseClient.class))
+                    .map(BaseClient::getVersionInfo)
+                    .toArray(CompletableFuture[]::new);
+
+            CompletableFuture.allOf(futures).join();
+
+            for (CompletableFuture<VersionInfo> cf : futures) {
+                if(!cf.get().getVersion().equals(getVersionString())) {
```
Comment (no2chem) [inline runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java]:
```java
if (cf.get().getVersion() == null) {
    log.error("Unexpected server version, server is too old to return version information");
}
```

**Comment by corfudb-performance** [comment]
Results automatically generated by _CorfuDB Benchmark Framework_ to assess the performance of this pull request for commit **edbf997**.

*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 10 threads, Disk mode
 
 An interactive dashboard with Pull Request Performance Metrics for ALL cluster types and numbers of threads in run, is available at:
 [Pull Request #879 Graphs ](https://snapshot.raintank.io/dashboard/snapshot/qqrvQYSrDdlo8nKuRTYttR0dROLoh9Jp)

File: runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java (hunk 1 above)
Comment (Maithem) [inline runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java]: Yep, not done yet.

**Comment by Maithem** [comment]
@no2chem ok, addressed all comments.

**Comment by Maithem** [comment]
test

**Comment by corfudb-performance** [comment]
Results automatically generated by _CorfuDB Benchmark Framework_ to assess the performance of this pull request for commit **7a02879**.

*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 10 threads, Disk mode
 
 An interactive dashboard with Pull Request Performance Metrics for ALL cluster types and numbers of threads in run, is available at:
 [Pull Request #879 Graphs ](https://snapshot.raintank.io/dashboard/snapshot/iFs82Bdlqy8icuq0Oby3eFl54EX1dzUr)

**Comment by Maithem** [comment]
@no2chem good to go ?

**Pull request merged**